src/components/LocalNav/LocalNav.tsx

```diff
@@ -1,108 +1,189 @@
+                    : (theme: Theme) => theme.palette.info.dark,
+                width: '0.5rem'
+              }}
+            />
+            <Grid item xs={7} sm={7} md={8} lg={8} xl={8}>
+              <ListItemText primary={topic.name} primaryTypographyProps={{ fontSize: 18 }} />
+            </Grid>
+            <Grid item xs={3} sm={3} md={3} lg={2} xl={2}>
+              {topicProgress && !isProgressLoading ? (
+                <ListItemText
+                  primary={
+                    <Fraction
+                      variant="body1"
+                      sx={{ fontSize: 16 }}
+                      numerator={topicProgress[0]}
+                      denominator={topicProgress[1]}
+                    />
+                  }
+                  primaryTypographyProps={{
+                    p: 0.25,
+                    borderRadius: 3,
+                    bgcolor: (theme: Theme) => theme.palette.info.light
+                  }}
+                  sx={{ textAlign: 'center' }}
+                />
+              ) : (
+                <Skeleton variant="text" width={'70%'} height={20} sx={{ ml: 2 }} />
+              )}
+            </Grid>
+          </Grid>
+        </ListItemButton>
+      </ListItem>
+    </Grid>
+  )
+}
 /**
  *  Local navigation component props.
- *  @prop {@link _useLearningPathTopic} - hook to get learning path topics
- *  @prop {@link _useLearningPathElement} - hook to get learning path elements
+ *  @prop {@link useLearningPathTopic} - hook to get learning path topics
  */
 export type LocalNavProps = {
   useLearningPathTopic?: (courseId: string) => { loading: boolean; topics: Topic[] }
-  useLearningPathElement?: (
-    topic: Topic,
-    courseId: string
-  ) => {
-    loadingElements: boolean
-    learningPaths: LearningPathElement | undefined
-  }
+}
+
+type RouteParams = {
+  courseId: string
+  topicId: string
 }
 
 /**
  * Local navigation component.
- * @param param - component props. The {@link LocalNavProps#useLearningPathTopic} and {@link LocalNavProps#useLearningPathElement} are optional.
+ * @param param - component props. The {@link LocalNavProps#useLearningPathTopic} are
+ *   optional.
  * @returns
+ * A JSX Element with the rendered local navigation.
  */
-const LocalNav = ({
-  useLearningPathTopic = _useLearningPathTopic,
-  useLearningPathElement = _useLearningPathElement
-}: LocalNavProps) => {
+const LocalNav = ({ useLearningPathTopic = _useLearningPathTopic }: LocalNavProps) => {
   const { t } = useTranslation()
-  const { courseId } = useParams() as { courseId: string }
-  const { loading, topics } = useLearningPathTopic(courseId)
+  const { courseId } = useParams<string>()
+  const { topicId } = useParams<string>()
+  const definedCourseId = courseId ?? "default"
+  const definedTopicId = topicId ?? "default"
+
+  const theme = useTheme()
+  const open = useMediaQuery(theme.breakpoints.up('lg'))
+  const [drawerHeight, setDrawerHeight] = useState(0)
```

Comment:
Bissle aufgeräumt für mehr Überblick
```suggestion
  // Default values
  const definedCourseId = courseId ?? "default"
  const definedTopicId = topicId ?? "default"

  // States
  const [drawerHeight, setDrawerHeight] = useState(0)

  // Hooks
  const { t } = useTranslation()
  const { courseId } = useParams<string>()
  const { topicId } = useParams<string>()
  const theme = useTheme()
  const open = useMediaQuery(theme.breakpoints.up('lg'))
  const { topics, loading: topicLoading } = useLearningPathTopic(definedCourseId)
  const { topicProgress, loading: isProgressLoading } = useLearningPathTopicProgress(definedCourseId, topics)
```